### PR TITLE
Improve error reporting when reading the config.

### PR DESCRIPTION
**Pull request by robewald**
It takes a while to figure out why you get an exception, when you have a syntax error in your config. This patch gives a better indication where there is a problem.

### CODE REVIEW

**Comment by d12frosted** [comment]
Looks pretty useful. Hope to see it in master.

File: OmniSharp/Configuration/PathReplacement.cs [hunk 1]
```diff
@@ -21,8 +21,30 @@ public static OmniSharpConfiguration Load(string configLocation, string clientMo
...
+		
+		int characc = 0;
+		for (int i = 0; i<lines.Length; i++) {
+		    int characc1 = characc + lines[i].Length;
+		    if (characc1 >= offset) {
+		    Console.WriteLine(configLocation + "(" + i + "," + (offset-characc) + "):"+lines[i]);
+		    break;
+		    }
+		}
+		Environment.Exit(1);
```
Comment (nosami) [inline OmniSharp/Configuration/PathReplacement.cs]: Could just throw the error here?

File: OmniSharp/Configuration/PathReplacement.cs [hunk 2]
```diff
@@ -21,8 +21,30 @@ public static OmniSharpConfiguration Load(string configLocation, string clientMo
...
+	    } catch (System.ArgumentException e)
+	    {
+		Console.WriteLine(e.Message);
+		const string pattern = @"\(([0-9]+)\)$";
+		int offset;
+		if (int.TryParse(Regex.Match(e.Message, pattern).Groups[1].Value, out offset))
+		{
+		string[] lines = config.Replace("\r\n","\n").Replace("\n\r","\n").Split('\n');
+		
+		int characc = 0;
```
Comment (nosami) [inline OmniSharp/Configuration/PathReplacement.cs]: what's characc?

**Comment by nosami** [comment]
indentation is all wrong :)

File: OmniSharp/Configuration/PathReplacement.cs (hunk 2 above)
Comment (robewald) [inline OmniSharp/Configuration/PathReplacement.cs]: A char accumulator.

**Comment by robewald** [comment]
@nosami, please check out the updated pull request.

**Pull request closed without merging**